src/Integration.Vsix/SonarLintDaemonIntegrationPackage.cs

```diff
@@ -0,0 +1,155 @@
+    [ProvideAutoLoad(UIContextGuids80.SolutionExists)]
+    [ProvideAutoLoad(UIContextGuids80.NoSolution)]
+    [SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1650:ElementDocumentationMustBeSpelledCorrectly", Justification = "pkgdef, VS and vsixmanifest are valid VS terms")]
+    public sealed class SonarLintDaemonIntegrationPackage : Package
+    {
+        private static readonly string DAEMON_HOST = "localhost";
+        private static readonly int DAEMON_PORT = 8050;
+
+        /// <summary>
+        /// SonarLintDaemonIntegrationPackage GUID string.
+        /// </summary>
+        public const string PackageGuidString = "6f63ab5a-5ab8-4a0d-9914-151911885966";
+
+        // these cannot be local variables, as they will get garbage collected
+        private static DTE dte;
+        private DocumentEvents documentEvents;
+
+        internal static SonarLintDaemonIntegrationPackage Instance { get; private set; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SonarLintDaemonIntegrationPackage"/> class.
+        /// </summary>
+        public SonarLintDaemonIntegrationPackage()
+        {
+            // Inside this method you can place any initialization code that does not require
+            // any Visual Studio service because at this point the package object is created but
+            // not sited yet inside Visual Studio environment. The place to do all the other
+            // initialization is the Initialize method.
+        }
+
+        #region Package Members
+
+        /// <summary>
+        /// Initialization of the package; this method is called right after the package is sited, so this is the place
+        /// where you can put all the initialization code that rely on services provided by VisualStudio.
+        /// </summary>
+        protected override void Initialize()
+        {
+            base.Initialize();
+
+            InitializeSonarLintDaemonFacade();
+
+            Instance = this;
+        }
+
+        #endregion
+
+        private void InitializeSonarLintDaemonFacade()
+        {
+            dte = (DTE)GetService(typeof(SDTE));
+            documentEvents = dte.Events.DocumentEvents;
+            documentEvents.DocumentSaved += DocumentSaved;
+        }
+
+        private void DocumentSaved(Document document)
+        {
+            if (document == null || document.Language != "JavaScript")
+            {
+                return;
+            }
+
+            RequestAnalysis(document.FullName);
+        }
+
+        public void RequestAnalysis(string path)
+        {
+            Analyze(path);
+        }
+
+        private async System.Threading.Tasks.Task Analyze(string path)
+        {
+            Channel channel = new Channel(string.Join(":", DAEMON_HOST, DAEMON_PORT), ChannelCredentials.Insecure);
+            var client = new StandaloneSonarLint.StandaloneSonarLintClient(channel);
+
+            InputFile inputFile = new InputFile();
```

Comment:
Note that the `,` on the last line of the initializer is not a mistake, it is ignored by the compiler, but it is a convenience in case you want to comment that line.